boards/derfmega256/include/eui_provider_params.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_deRFmega256
+ * @{
+ *
+ * @file
+ * @brief       EUI providers found on the board
+ *
+ * @author      Alexander Chudov <chudov@gmail.com>
+ */
+#ifndef EUI_PROVIDER_PARAMS_H
+#define EUI_PROVIDER_PARAMS_H
+
+#include "net/eui64.h"
+#include "periph/eeprom.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    MAC configuration
+ *          Offset of the MAC address in the EEPROM
+ */
+#define EEPROM_MAC_ADDR (0x1fe4)
+
+/**
+ * @brief    Constant in EEPROM provides a EUI-64, this is also printed on the board
+ */
+static inline int _eeprom_mac_get_eui64(uint8_t index, eui64_t *addr)
+{
+    (void)index;
+
+    if (eeprom_read(EEPROM_MAC_ADDR, addr, sizeof(eui64_t)) != sizeof(eui64_t)) {
+        return -1;
+    }
+
+    addr->uint64.u64 = byteorder_htonll(addr->uint64.u64).u64;
```

Comment:
I wonder why the EUI-64 wasn't stored in network byte order to avoid this back and forth conversion. But anyway, existing users will get unhappy if we just flip the byte order here.

Reply:
probably because ATMega256RFR2 uses little endian for the MAC address.
`at86rf2xx` takes care of converting EUI-64 from network byte order to host byte order for the radio.

Reply:
Which also means `at86rf2xx_get_addr_long` is buggy :weary: